### PR TITLE
Added page factory design pattern

### PR DESCRIPTION
**Pull request by osandadeshan**
Page-Factory is a cross-platform framework that allows you to run tests on all popular browsers, because Selenium WebDriver is used to run them.

### CODE REVIEW

**Comment by iluwatar** [comment]
@osandadeshan since this change comes from outside our issue list, can you say where this testing pattern has been best described?

**Comment by osandadeshan** [comment]
@iluwatar It is actually a test automation design pattern vastly used in the industry.
https://www.browserstack.com/guide/page-object-model-in-selenium

File: page-factory/pom.xml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    <packaging>pom</packaging>
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-assembly-plugin</artifactId>
+            </plugin>
+        </plugins>
+    </build>
+</project>
```
Comment (ohbus) [inline page-factory/pom.xml:38]: add an empty line at the end of the file

**Review by ohbus — CHANGES_REQUESTED** [review]

**Review by ohbus — COMMENTED** [review]
I cannot seem to understand one thing which is everywhere in the web _(including the reference that you have provided in the comment)_ mentions the pattern to be called as **Page Object Model** whereas you have used the term **Page Factory** design pattern.

Please clarify on the above point to help me understand you better.

Thank you.

**Comment by ohbus** [comment]
> I cannot seem to understand one thing which is everywhere in the web _(including the reference that you have provided in the comment)_ mentions the pattern to be called as **Page Object Model** whereas you have used the term **Page Factory** design pattern.
> 
> Please clarify on the above point to help me understand you better.
> 
> Thank you.

@osandadeshan  What I understand from the above perspective that **Page Factory Design Pattern** is **Enhanced POM**
but on which ground.

**Comment by osandadeshan** [comment]
Selenium's PageFactory, is simply an annotation that serves to delegate to Selenium the instantiation of some WebElement objects you might want to have on an object that represents a web page or a component.

As mentioned on the [Selenium documentation](https://github.com/SeleniumHQ/selenium/wiki/PageObjects):

**`There is a PageFactory in the support package that provides support for this pattern, and helps to remove some boiler-plate code from your Page Objects at the same time.`**

PageFactory serves to make things more succinct, it does not implement the Page Object design pattern out-of-the-box.

**Comment by tomaszwojciechowski** [comment]
PageFactory is not a design pattern. The design pattern is Page Object, while PageFactory is a class in the Selenium framework that supports the Page Object pattern by removing the boilerplate code from the Page Object classes, instantiating those classes and setting up a lazy proxy for WebElement and List \<WebElement\> fields.

**Comment by iluwatar** [comment]
I agree with @tomaszwojciechowski. We already have Page Object pattern, see https://github.com/iluwatar/java-design-patterns/issues/229.

**Pull request closed without merging**